### ISSUE TITLE
[Feature Request] Add a `LinkHint` to mimic the middle-click

### ISSUE THREAD

**Issue by MahdiNazemi** [gdh1995/vimium-c#891]
**What command or commands**

`LinkHints`

**What should a feature do**

Currently, `LinkHints.activateOpenInNewTab` opens a new tab in the same container as the link hint. I use the [Temporary Containers](https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/) extension and have set my Isolation settings to open a new container on middle-click if the target belongs to a different domain or subdomain.

Is it possible to add a `LinkHint` that mimics the middle-click, so I can map <kbd>F</kbd> to it?

Something like this would be consistent with how right click can be defined in Vimium C:
```
unmap F
map F LinkHints.activateOpenInNewTab button="middle"
```

**Browser and OS**

- Browser name: Firefox 111.0
- OS name: macOS Ventura 13.2.1

**Comment by gdh1995** [comment]
Well, I'll add it in a next version. In currently v1.99.96/97, you may write:
```
map F LinkHints.activateHover $then="mid_click"
run <v-sim_mid> dispatchEvent#button=1&type=$s mask
run <v-mid_click> sim_mid#mousedown+sim_mid#mouseup
```

**Comment by MahdiNazemi** [comment]
> ```
> map F LinkHints.activateHover $then="mid_click"
> run <v-sim_mid> dispatchEvent#button=1&type=$s mask
> run <v-mid_click> sim_mid#mousedown+sim_mid#mouseup
> ```

This highlights the link for me but does not open it in.

**Comment by gdh1995** [comment]
During my test it does send messages to the extension, but, I also found that no tab was opened. Not sure why

**Comment by MahdiNazemi** [comment]
My knowledge in this area is very limited, but when trying to simulate a middle-click, do we dispatch an event where [`isTrusted`](https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted) is false and the browser ignores that event?

**Comment by gdh1995** [comment]
I just viewed source code of the extension `Temporary Containers`, and I found:
* it doesn't use `isTrusted`
* it never "blocks" click events, while it just learn info from every click, and then hooks Firefox's "open a new tab" actions
* however, Vimium's simulated click events won't trigger a real "open a new tab" action.

I have to add more parameters to express such a use case.

**Comment by gdh1995** [comment]
Hello, I find a way to work with the extension in the current version v1.99.97 (also v1.99.96) of Vimium C:
```
map F LinkHints.activateHover $then="$D+mid_click"
run <v-mid_click> sim_mid#mousedown+$D+sim_mid#mouseup+$D+lh_open
run <v-sim_mid> dispatchEvent#button=1&type=$s mask
map <v-lh_open> LinkHints.click mode="incognito" incognito=false
```

**Comment by gdh1995** [comment]
In v1.99.98, the mapping will be simplified to:
```
map F LinkHints.activate button=1 $then="$D+lh_open"
map <v-lh_open> LinkHints.click mode="open-directly"
```

**Comment by MahdiNazemi** [comment]
> ```
> map F LinkHints.activateHover $then="$D+mid_click"
> run <v-mid_click> sim_mid#mousedown+$D+sim_mid#mouseup+$D+lh_open
> run <v-sim_mid> dispatchEvent#button=1&type=$s mask
> map <v-lh_open> LinkHints.click mode="incognito" incognito=false direct="hovered"
> ```

Thank you. Works like a charm!

**Comment by gdh1995** [comment]
Well, there's a bug in the previous mapping - when a selected element has no valid URL, then Vimium C will open copied text in your clipboard...

There seems no easy way to fix it in up to v1.99.97. So you have to wait for weeks.

In a future version (v1.99.98+), a better mapping will be:
```
map F LinkHints.activate button=1 $then="$D+lh_open"
map <v-lh_open> LinkHints.click mode="open-directly" \
  direct="hovered" sed="s@^$@vimium://run/blank@,act"
```

**Comment by MahdiNazemi** [comment]
> ```
> map F LinkHints.activateHover $then="$D+mid_click"
> run <v-mid_click> sim_mid#mousedown+$D+sim_mid#mouseup+$D+lh_open
> run <v-sim_mid> dispatchEvent#button=1&type=$s mask
> map <v-lh_open> LinkHints.click mode="incognito" incognito=false direct="hovered"
> ```

Is it possible to have `F` open the link in the background, as the middle click does? The current mapping changes focus to the opened tab.

**Comment by gdh1995** [comment]
replace the last with
```
map <v-lh_open> LinkHints.click mode="incognito" incognito=false direct="hovered" reuse="new-bg"
```

**Comment by MahdiNazemi** [comment]
The four lines implementing middle click simulation perform a duckduckgo search if the activated link hint is not designed to be opened in a new window, e.g., the StackOverflow upvote button.

**Comment by gdh1995** [comment]
> Well, there's a problem in the previous mapping - when a selected element has no valid URL, then Vimium C will open copied text in your clipboard...
> 
> There seems no easy way to fix it in up to v1.99.97. So you have to wait for weeks.
> 
> In a future version (v1.99.98+), I'll fix this bug and then a shorter mapping will be:
> 
> ```
> map F LinkHints.activate button=1 $then="$D+lh_open"
> map <v-lh_open> LinkHints.click mode="open-directly" direct="hovered"
> ```

See this. You may need to wait for some weeks.

**Comment by MahdiNazemi** [comment]
The following mapping simulates middle-click and opens the link in a new background tab:

```vim
map F LinkHints.activate button=1 $then="$D+tab_bg"
map <v-tab_bg> LinkHints.click mode="open-directly" direct="hovered" reuse="new-bg"
```
